src/compile/util.ts

```diff
@@ -52,3 +55,84 @@ export function applyMarkConfig(marksProperties, model: Model, propsList: string
     }
   });
 }
+
+
+/**
+ * Builds an object with format and formatType properties.
+ *
+ * @param format explicitly specified format
+ */
+export function formatMixins(model: Model, channel: Channel, format: string) {
+  const fieldDef = model.fieldDef(channel);
+
+  let def: any = {};
+
+  switch (fieldDef.type) {
+    case QUANTITATIVE:
+      def.formatType = 'number';
+      break;
+    case TEMPORAL:
+      def.formatType = 'time';
+      break;
+  }
+
+  if (format !== undefined) {
+    def.format = format;
+  } else {
+    switch (fieldDef.type) {
+      case QUANTITATIVE:
+        def.format = model.numberFormat(channel);
```

Comment:
I guess `numberFormat` is only used here -- remove `model.numberFormat()` and move the logic here?